flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -924,5 +924,10 @@ public long getDefaultBufferTimeout() {
         public ReadableConfig getGraphGeneratorConfig() {
             return config;
         }
+
+        @Override
+        public Collection<Integer> transform(Transformation<?> transformation) {
```

Comment:
That is definitely not the part I am most proud of :) Unfortunately, I need a way in a `SinkTransformationTranslator` to enqueue new transformations for the translation to support the dynamic expansion in case a sink uses a custom topology. 

The only thing exposed is the `Context` in `translateForStreaming` and `translateForBatch`.